test/Sprites/ArtWriter.test.cpp

```diff
@@ -21,3 +23,40 @@ TEST(ArtWriter, BlankFilename)
 {
 	EXPECT_THROW(ArtFile::Write("", ArtFile()), std::runtime_error);
 }
+
+TEST(ArtWriter, scanLineByteWidth)
+{
+	ArtFile artFile;
+	artFile.palettes.push_back(Palette());
+
+	ImageMeta imageMeta;
+	imageMeta.width = 10;
+	imageMeta.scanLineByteWidth = 22;
+	imageMeta.paletteIndex = 0;
+
+	artFile.imageMetas.push_back(imageMeta);
+
+	std::string filename = "Sprites/data/test.prt";
+
+	EXPECT_THROW(ArtFile::Write(filename, artFile), std::runtime_error);
+
+	XFile::DeletePath(filename);
+}
+
+TEST(ArtWriter, paletteIndex) 
```

Comment:
I'm also uncertain what this is trying to test, and would appreciate a more detailed test name here. The only difference I spot is the `scanlineByteWidth`, which is now a multiple of 4.

Reply:
This is testing to ensure an error is thrown when writing if an ImageMeta palette index is outside the range of palettes being saved. I'll update to a little more descriptive test name. Let me know if it is still too vague.

Reply:
Ahh, ok, I missed some of the detail there.

Hmm, if ArtFile is a class with encapsulation, you can error check when setting the data, rather than when writing the data. Checking a palette index can be done long before the data is written. That moves the error reporting closer to the point that caused the error. It also means simpler testing as you don't have to generate and delete a temporary file.

For now though, I think a better name would suffice.

I think test names should follow the advice for "Verify" methods. They should describe the property being tested, rather than a noun phrase that has something tested about it. For example: `ScanlineByteWidthNonAlignedThrows`, or `PaletteIndexOutOfBoundsThrows`, or `PaletteColorsNotSwappedAfterWrite`.